spec/host_inventory/linux/interface_spec.rb

```diff
@@ -0,0 +1,271 @@
+    link/ether fe:54:00:7b:69:9b brd ff:ff:ff:ff:ff:ff
+    inet6 fe80::fc54:ff:fe7b:699b/64 scope link 
+       valid_lft forever preferred_lft forever
+EOH
+
+
+## Output from 'ip -oneline addr show'
+str[:interfaces] = <<-EOH
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN 
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+    inet6 ::1/128 scope host 
+       valid_lft forever preferred_lft forever
+2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
+    link/ether 62:24:49:e8:16:b2 brd ff:ff:ff:ff:ff:ff
+     inet 192.168.1.116/24 brd 192.168.1.255 scope global eth0
+     inet 10.0.0.1/24 scope global eth0-foo 
+     inet 10.0.0.2/24 scope global secondary eth0-foo
+     inet6 2001:db8::16b4/64 scope global temporary dynamic 
+        valid_lft 86395sec preferred_lft 14395sec
+     inet6 2001:db8::16b3/64 scope global deprecated dynamic 
+        valid_lft 86395sec preferred_lft 0sec
+     inet6 2001:db8::16b2/64 scope global dynamic 
+        valid_lft 86395sec preferred_lft 14395sec
+     inet6 fe80::6224:49ff:fee8:16b2/64 scope link 
+        valid_lft forever preferred_lft forever
+3: venet0: <BROADCAST,POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN 
+    link/void 
+    inet 127.0.0.1/32 scope host venet0
+    inet 192.0.2.1/32 brd 192.0.2.1 scope global venet0:0
+    inet6 2001:db8::1/128 scope global 
+       valid_lft forever preferred_lft forever
+    inet6 ::2/128 scope global 
+       valid_lft forever preferred_lft forever
+57: usb0: <BROADCAST,MULTICAST> mtu 1500 qdisc pfifo_fast state DOWN qlen 1000
+    link/ether 02:80:37:ec:02:00 brd ff:ff:ff:ff:ff:ff
+EOH
+
+str[:rhel6] = <<-EOH
+Kernel 2.6.32-573.8.1.el6.x86_64
+
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+    inet6 ::1/128 scope host
+       valid_lft forever preferred_lft forever
+2: eth0: <BROADCAST,MULTICAST,SLAVE,UP,LOWER_UP> mtu 1500 qdisc mq master bond0 state UP qlen 1000
+    link/ether 33:44:55:96:70:1c brd ff:ff:ff:ff:ff:ff
+3: eth1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
+    link/ether 33:44:55:96:70:1d brd ff:ff:ff:ff:ff:ff
+4: eth2: <BROADCAST,MULTICAST,SLAVE,UP,LOWER_UP> mtu 1500 qdisc mq master bond0 state UP qlen 1000
+    link/ether 33:44:55:96:70:1c brd ff:ff:ff:ff:ff:ff
+5: eth3: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
+    link/ether 33:44:55:96:70:1f brd ff:ff:ff:ff:ff:ff
+6: bond0: <BROADCAST,MULTICAST,MASTER,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP
+    link/ether 31:44:55:96:70:1c brd ff:ff:ff:ff:ff:ff
+    inet 10.3.5.172/26 brd 10.3.5.191 scope global bond0
+   inet6 fe80::56:44:55ff:fe96:701c/64 scope link
+       valid_lft forever preferred_lft forever
+EOH
+
+
+describe Specinfra::HostInventory::Network do
```

Comment:
Thanks for the pointer. 

I have actually changed some of the things already which can be seen in the my branch https://github.com/uroesch/specinfra/tree/wip/host_inventory-network 
I'm currently testing in our production environment and hope to have it ready soon.